dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/db/DbType.kt

```diff
@@ -96,4 +194,213 @@ public abstract class DbType(public val dbTypeInJdbcUrl: String) {
         }
         return connection
     }
+
+    /**
+     * Extracts a value from the ResultSet for the given column.
+     * This method can be overridden by custom database types to provide specialized parsing logic.
+     *
+     * @param [rs] the ResultSet to read from
+     * @param [columnIndex] zero-based column index
+     * @param [columnMetadata] metadata for the column
+     * @param [kType] the Kotlin type for this column
+     * @return the extracted value, or null
+     */
+    public open fun extractValueFromResultSet(
+        rs: ResultSet,
+        columnIndex: Int,
+        columnMetadata: TableColumnMetadata,
+        kType: KType,
+    ): Any? =
+        try {
+            rs.getObject(columnIndex + 1)
+            // TODO: add a special handler for Blob via Streams
+        } catch (_: Throwable) {
+            // TODO: expand for all the types like in generateKType function
+            if (kType.isSupertypeOf(String::class.starProjectedType)) {
```

Comment:
try to avoid `::class`, it's way heavier than `typeOf<String>()`. Also, don't you mean `subType`? instead of `superType`?